default.nix

```diff
@@ -6,72 +6,126 @@ with pkgs;
 
 let
   root = ./.;
-  mainnet = {
+  mainnet = rec {
     rev = "94f779a7";
     sha256 = "16lxilng5q8fr2ll6h4hf7wlvac6nmw4cx10cbgzj5ks090bl97r";
     patchFile = ./nix/fix-mainnet.patch;
+    protoName = "005_PsBabyM1";
+    binarySuffix = builtins.replaceStrings ["_"] ["-"] protoName;
   };
-  babylonnet = {
+  babylonnet = rec {
     rev = "b8731913";
     sha256 = "1pakf1s6bg76fq42mb8fj1immz9g9wwimd522cpx8k28zf0hkl5i";
     patchFile = ./nix/fix-babylonnet.patch;
+    protoName = "005_PsBabyM1";
+    binarySuffix = builtins.replaceStrings ["_"] ["-"] protoName;
   };
-  tezos-client-static-mainnet = import ./nix/static.nix mainnet;
-  tezos-client-static-babylonnet = import ./nix/static.nix babylonnet;
-  binary-mainnet = "${tezos-client-static-mainnet}/bin/tezos-client";
-  binary-babylonnet = "${tezos-client-static-babylonnet}/bin/tezos-client";
-  packageDesc-mainnet = {
-    project = "tezos-client-mainnet";
-    version = toString timestamp;
-    bin = binary-mainnet;
-    arch = "amd64";
-    license = "MPL-2.0";
-    dependencies = "";
-    maintainer = "Serokell https://serokell.io";
-    licenseFile = "${root}/LICENSES/MPL-2.0.txt";
-    description = "CLI client for interacting with tezos blockchain";
-    gitRevision = mainnet.rev;
-    branchName = "mainnet";
-  };
+  tezos-static-mainnet = import ./nix/static.nix mainnet;
+  tezos-static-babylonnet = import ./nix/static.nix babylonnet;
 
-  packageDesc-babylonnet = packageDesc-mainnet // {
-    project = "tezos-client-babylonnet";
-    bin = binary-babylonnet;
-    gitRevision = babylonnet.rev;
-    branchName = "babylonnet";
-  };
+  packDirectory = archiveName: pathToPack:
+    stdenv.mkDerivation rec {
+      name = "${archiveName}.tar.gz";
+      phases = "archivePhase";
+      nativeBuildInputs = [ gnutar ];
+      archivePhase = ''
+        mkdir -p ${archiveName}
+        cp -a ${pathToPack}/. ${archiveName}/
+        tar -cvzf ${name} ${archiveName}
+        mkdir -p $out
+        cp ${name} $out/
+      '';
+    };
+
+  mainnet-binaries = packDirectory "tezos-mainnet-binaries-${mainnet.rev}"
+    "${tezos-static-mainnet}/bin";
+  babylonnet-binaries =
+    packDirectory "tezos-babylonnet-binaries-${babylonnet.rev}"
+    "${tezos-static-babylonnet}/bin";
+
+  mkPackageDescs = { executableName, binPath, description }@executableDesc:
+    let
+      mainnetDesc = {
+        inherit description;
+        bin = "${tezos-static-mainnet}/${binPath}";
+        gitRevision = mainnet.rev;
+        project = "${executableName}-mainnet";
+        version = toString timestamp;
```

Comment:
wait, why timestamp? If you must, grab it from git somehow maybe?

Reply:
we came up to an agreement, that we are going to use a timestamp as a version since `tezos-*` stuff lack any kind of versioning

Reply:
and not the timestamp from the tezos git revision, but the time, when static binary/package was built

Reply:
to be more specific, timestamp of package building, not the revision timestamp

Reply:
Poor deterministic nix build, tainted by human concepts such as time.